packages/web/src/common/store/queue/sagas.ts

```diff
@@ -265,6 +273,32 @@ export function* watchPlay() {
   })
 }
 
+// Fetches more lineup tracks if available. This is needed for cases
+// where the user hasn't scrolled through the lineup.
+function* fetchLineupTracks() {
+  const source = yield* select(getSource)
+  if (!source) return
+
+  const lineupEntry = lineupRegistry[source]
+  if (!lineupEntry) return
+
+  const currentArtist = yield* select(getCurrentArtist)
+  const lineup = yield* select(lineupEntry.selector, currentArtist?.handle)
+
+  if (lineup.hasMore) {
+    const offset = lineup.entries.length + lineup.deleted + lineup.nullCount
```

Comment:
is there any danger of this getting out of sync? I think we should be good since it's based purely off of the state, but we could also calculate this in the reducer just to be safe

Reply:
so i copied this from the lineup, so yes there is a chance of the logic getting out of sync, if we forget to change in both places. but hoping from the perspective of redux state, both sides (ui and saga) can work together effectively here.